src/main/desktop_settings.ts

```diff
@@ -0,0 +1,73 @@
+import { EventEmitter } from 'events'
+import { promisify } from 'util'
+import { getLogger } from '../shared/logger'
+import { DesktopSettingsType } from '../shared/shared-types'
+import { getDefaultState } from '../shared/state'
+import appConfig from './application-config'
+import debounce from 'debounce'
+const log = getLogger('main/state')
+
+const SAVE_DEBOUNCE_INTERVAL = 1000
+
+class PersistentState extends EventEmitter {
+  constructor() {
+    super()
+  }
+
+  private inner_state: null | DesktopSettingsType = null
+
+  get state(): Readonly<DesktopSettingsType> {
+    if (this.inner_state == null) {
+      throw new Error('Can not access persistent state before initialisation')
+    }
+    return this.inner_state
+  }
+
+  async load() {
+    const default_state = getDefaultState()
+    let saved: Partial<DesktopSettingsType> = {}
+    try {
+      saved = (await promisify(cb =>
+        appConfig.read(cb)
+      )()) as DesktopSettingsType
+      // validate&fix saved state
+      if (typeof saved.lastAccount !== 'number') {
+        saved.lastAccount = undefined
+      }
+    } catch (error) {
+      log.debug(error)
+      log.info('Missing configuration file. Using default values.')
+    }
+    this.inner_state = Object.assign(default_state, saved)
+  }
+
+  mutate(state: Partial<DesktopSettingsType>) {
```

Comment:
Not too happy about mutate, would prefer a simple update/set.

Reply:
just naming?